### DISCUSSION TITLE
Nestjs create RequestContext ?

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#4718] [Q&A] [answered]
I saw it in document, but I don't know how to control this behavior in NestJs ?
How can I do that ?

```Typescript
// `orm.em` is the global EntityManager instance

app.use((req, res, next) => {
  // calls `orm.em.fork()` and attaches it to the async context
  RequestContext.create(orm.em, next);
});

app.get('/', async (req, res) => {
  // uses fork from the async context automatically
  const authors = await orm.em.find(Book, {});
  res.json(authors);
});
```

**Comment by B4nan** [comment, marked as answer]
This is handled automatically via https://github.com/mikro-orm/nestjs, it registers the middleware for your.

**Reply by dungkaka** [reply]
And I can control this behavior through "registerRequestContext: true | false" in config, isn't it ?

**Reply by B4nan** [reply]
Yes, you can disable the middleware via `registerRequestContext` if you want to handle the context management yourself, its an option of that nest adapter.

You can as well configure the ORM to check your own callback when retrieving the context, this way you can bring your own `AsyncLocalStorage`:

https://mikro-orm.io/docs/usage-with-nestjs#using-asynclocalstorage-for-request-context

**Reply by dungkaka** [reply]
I got it, thank you so much !
